core/src/main/java/org/mskcc/oncotree/utils/TumorTypesUtil.java

```diff
@@ -88,15 +90,58 @@ public static Properties getProperties() {
         return filtered;
     }
 
-    public static InputStream getTumorTypeInputStream() {
-        Properties properties = getProperties();
-        InputStream inputStream = null;
+    public static InputStream getTumorTypeInputStream(Map<String, TumorType> tumorTypes) {
+        String tsvAsString = getTsvFromTumorTypes(tumorTypes);
         try {
-            inputStream = new FileInputStream(properties.getProperty("tumor_type_file_path"));
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
+            return new ByteArrayInputStream(tsvAsString.getBytes("UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+           e.printStackTrace();
+        }
+        return null;
+    }
+
+    private static String getTsvFromTumorTypes(Map<String, TumorType> tumorTypes) {
+        List<String> rows = new ArrayList<>();
+        rows.add(TSV_HEADER);
+        // could hard code this instead of for loop
+        for (String code : tumorTypes.keySet()) {
+            TumorType tumorType = tumorTypes.get(code);
+            // skip the root node, "TISSUE". Just add it's children
+            Map<String, TumorType> children = tumorType.getChildren();
+            for (String childCode : children.keySet()) {
+                addTumorTypeToRows(children.get(childCode), rows, new ArrayList<String>());
+            }
+        }
+        return StringUtils.join(rows, "\n");
+    }
+
+    private static void addTumorTypeToRows(TumorType tumorType, List<String> rows, List<String> parents) {
+        List<String> row = new ArrayList<>();
+        String displayName = StringUtils.defaultString(tumorType.getName()).trim() + " (" + StringUtils.defaultString(tumorType.getCode()).trim() + ")";
+        row.addAll(parents);
+        row.add(displayName);
+
+        // Need to pad for the primary - quinternary columns if all parents are not present
+        for (int i = 4; i > parents.size(); i--) {
+            row.add("");
+        }
+
```

Comment:
In the future, we may have oncotrees which go deeper than 5 children. We should probably add a check in this function to see if that is the case and to throw an exception if we see a node with depth > 5.